calcom/cal.com
# feat: Messaging Bus Implementation using AWS SQS OSSHack Challenge

Fixes for issue 12556: https://github.com/calcom/cal.com/issues/12556

## What does this PR do?

We refactored the code in order to support using queueing with AWS SQS in order to improve the performance, reliability, and scalability of the code. Functionality for sending emails, webhook messages, and calendar events were moved to a separate worker process that consumes these events off the SQS queue. We chose SQS because it is easy to use, which is important for the open source community, it is reliable, scalable, and integrates well with cal.com's current stack as they already leverage AWS S3.

Notably, this code is fully backward compatible and relies on new environment variables as documented in .env.example to trigger this new queueing functionality. The only dependency required for this change is @aws-sdk/client-sqs.


## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?



- Are there environment variables that should be set?
The following environment variables need to be set as according to the documentation in .env.example. To trigger this new functionality, AWS_SQS_CONSUMER must be set to true. All other environment variables listed below. Notably, we had issues integrating environment variables with our separate worker process, so we were forced to hardcode our values in awsSqsClient.ts. In addition to updating the below environment variables, please also change the variables in worker.tsx where relevant (there are comments showing which lines correspond to which environment variables).
AWS_SQS_CONSUMER=true
AWS_SQS_URL=
AWS_REGION=
AWS_ACCESS_KEY=
AWS_SECRET_KEY=
AWS_WEBHOOK_SECRET=

- What are the minimal test data to have?
Integration with send grid and google calendar

- Any other important info that could help to test that PR
To test our code, we booked calendar events and checked that the critical paths followed for sending emails and generating/updating calendar events with the worker application were the same as the existing code. We confirmed this functionality still works out of the box (for backward compatibility) and likewise works with our new AWS flags enabled.

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
